### PR TITLE
Refactor (handler part5) Extract MLX decode helpers into DecodeMixin with unit coverage

### PR DESCRIPTION
**Pull request by 1larity**
## Summary

Continues decomposition of `handler.py` by extracting MLX decode logic into a dedicated mixin.

This follows the established functional-decomposition strategy and reduces monolithic responsibilities inside `AceStepHandler`.

---

### New module
- `decode.py`

### Updated wiring
- Updated handler facade exports:
  - `__init__.py`
- Updated `AceStepHandler` inheritance to include:
  - `DecodeMixin` (in `handler.py`)

### Removed from `handler.py`
- `_mlx_vae_decode`
- `_mlx_decode_single`

Method bodies were moved intact into the mixin.

---

## Behavior / Correctness Notes

- Decode behavior is preserved via mixin composition.
- Added safety guard in `_mlx_decode_single` to prevent overlap-trim boundary collapse resulting in empty or invalid slices.
- Added explicit host-attribute validation in `DecodeMixin` for required runtime dependencies (e.g. `mlx_vae`, `disable_tqdm`).
- Clarified docstring terminology:
  - Explicitly documents NCL input
  - Clarifies internal NLC conversion

No feature changes intended — move-only decomposition plus safety hardening.

---

## Tests

Added focused unit tests:

- `decode_test.py`

Validation:

```bash
python -m unittest acestep.core.generation.handler.decode_test
python -m unittest acestep.core.generation.handler.diffusion_test

Validation for this slice is unit-test only.

No Apple hardware available in this environment.

No real MLX runtime or Apple Silicon integration verification performed in this pass.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Moved latent-to-audio decoding and diffusion execution into dedicated, reusable handler components and exposed them as part of the public handler surface to simplify generation workflows.

* **Tests**
  * Added extensive unit tests covering decoding (including tiled/overlap decoding), diffusion execution, input validation, error paths, and dtype/device handling to improve reliability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: acestep/handler.py [hunk 1]
```diff
@@ -43,15 +43,21 @@
...
+)
 from acestep.dit_alignment_score import MusicStampsAligner, MusicLyricScorer
 from acestep.gpu_config import get_gpu_memory_gb, get_global_gpu_config, get_effective_free_vram_gb
 
 
 warnings.filterwarnings("ignore")
 
 
-class AceStepHandler(InitServiceMixin, LoraManagerMixin, ProgressMixin):
+class AceStepHandler(DecodeMixin, DiffusionMixin, InitServiceMixin, LoraManagerMixin, ProgressMixin):
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/handler.py:60]: Acceptable, this is the module being decomposed step-by-step

File: acestep/core/generation/handler/decode.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import math
+from typing import Any, Callable, Optional
+
+import torch
+from loguru import logger
+from tqdm import tqdm
+
+
+class DecodeMixin:
+    """Mixin containing latent-to-audio decode helpers."""
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/core/generation/handler/decode.py:12]: Acceptable, hard limit is 200 loc

File: acestep/core/generation/handler/diffusion_test.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                encoder_hidden_states=x,
+                encoder_attention_mask=torch.ones(1, 2, dtype=torch.int64),
+                context_latents=torch.randn(1, 4, 3),
+                src_latents=torch.randn(1, 2, 3),
+                seed=1,
+            )
+
+
+if __name__ == "__main__":
+    unittest.main()
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/core/generation/handler/diffusion_test.py:198]: Rerun test, displayed code is old version, docstrings are present in latest
